edalize/libero.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import os.path
+import os
+import platform
+import subprocess
+import re
+import xml.etree.ElementTree as ET
+from functools import partial
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+
+class Libero(Edatool):
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            return {'description': "The Libero backend supports Microsemi Libero to build systems and program the FPGA",
+                    'members': [
+                        {'name': 'family',
+                         'type': 'String',
+                         'desc': 'FPGA family (e.g. PolarFire)'},
+                        {'name': 'die',
+                         'type': 'String',
+                         'desc': 'FPGA device (e.g. MPF300TS)'},
+                        {'name': 'package',
+                         'type': 'String',
+                         'desc': 'FPGA package type (e.g. FCG1152)'},
+                        {'name': 'speed',
+                         'type': 'String',
+                         'desc': 'FPGA speed rating (e.g. -1)'},
+                        {'name': 'dievoltage',
+                         'type': 'String',
+                         'desc': 'FPGA die voltage (e.g. 1.0)'},
+                        {'name': 'range',
+                         'type': 'String',
+                         'desc': 'FPGA temperature range (e.g. IND)'},
+                        {'name': 'defiostd',
+                         'type': 'String',
+                         'desc': 'FPGA default IO std (e.g. "LVCMOS 1.8V"'},
+                        {'name': 'hdl',
+                         'type': 'String',
+                         'desc': 'Default HDL (e.g. "VERILOG"'}
+                    ]
+                    }
+
+    argtypes = ['vlogdefine', 'vlogparam']
+
+    tool_options_defaults = {
+        'speed': '-1',
+        'dievoltage': '1.0',
+        'range': 'IND',
+        'defiostd': 'LVCMOS 1.8V',
+        'hdl': 'VERILOG',
+    }
+
+    def _set_tool_options_defaults(self):
+        for key, default_value in self.tool_options_defaults.items():
+            if not key in self.tool_options:
+                logger.info("Set Libero tool option %s to default value %s"
+                            % (key, str(default_value)))
+                self.tool_options[key] = default_value
+
+    """ Initial setup of the class
+
+    This calls the parent constructor, but also identifies whether
+    the current system is using a Standard or Pro edition of Quartus.
+    """
+
+    def __init__(self, edam=None, work_root=None, eda_api=None):
```

Comment:
We can remove the constructor here. The super class constructor will be called automatically for us